db/config.cc

```diff
@@ -840,6 +840,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , max_memory_for_unlimited_query_hard_limit(this, "max_memory_for_unlimited_query_hard_limit", "max_memory_for_unlimited_query", liveness::LiveUpdate, value_status::Used, (uint64_t(100) << 20),
             "Maximum amount of memory a query, whose memory consumption is not naturally limited, is allowed to consume, e.g. non-paged and reverse queries. "
             "This is the hard limit, queries violating this limit will be aborted.")
+    , twcs_max_window_count(this, "twcs_max_window_count", liveness::LiveUpdate, value_status::Used, 50,
+            "The maximum number of compaction windows allowed when making use of TimeWindowCompactionStrategy. A setting of 0 effectively disables the restriction.")
```

Comment:
Note that your code above doesn't actually enforce this in the general case - only when default TTL is enabled.

Reply:
That's correct, because tables without a TTL defined can either have a bound (provided you use `USING TTL`) or unbound window size. Ideally, one would want to always forbid creation of tables without a TTL defined - **BUT**, this alone is already enough to break tools such as cassandra-stress. And the only way to tell cassandra-stress to create a table with a defined TTL is to provide an user profile.

That said, instead of failing everything altogether (and a bunch of dtests and units), we print a warning (in another commit) other than immediately refusing the creation of the table. For production settings, however, we should enforce it.

Reply:
What I meant was that the option string isn't 100% accurate because it doesn't mention that this restriction only applies to tables with a default ttl. But this is a minor quibble, let's not delay the merge for this.